lm_eval/api/task.py

```diff
@@ -204,20 +204,18 @@ def __init__(
         self._fewshot_docs = None
         self._instances = None
 
-        self._config = TaskConfig(**config) if config else TaskConfig()
-
-        if not hasattr(self, "_filters"):
-            self._filters = []
-            for name, components in self._config.get(
-                "filters", [["none", [["take_first", None]]]]
-            ):
-                filter_pipeline = build_filter_ensemble(name, components)
-                self._filters.append(filter_pipeline)
-
-        self.sampler = samplers.Sampler(
-            list(self.fewshot_docs()), self, rnd=random.Random(1234)
+        self._config = (
+            TaskConfig(
+                {
+                    **config,
+                }
+            )
+            if config
+            else TaskConfig()
         )
 
+        self._filters = [build_filter_ensemble("none", [["take_first", None]])]
```

Comment:
this breaks the case where a task has several different filter pipelines applied

Reply:
But for manually written tasks, any filtering will be implemented by the user, right?

The filter pipeline is still intact for `ConfigurableTask`.